homeassistant/components/network/network.py

```diff
@@ -0,0 +1,237 @@
+            adapter["enabled"] = True
+            return
+
+    _enable_adapters(adapters, _adapters_with_exernal_addresses(adapters))
+
+
+def _adapters_with_exernal_addresses(adapters: list[Adapter]) -> list[str]:
+    """Enable all interfaces with an external address."""
+    return [
+        adapter["name"]
+        for adapter in adapters
+        if _adapter_has_external_address(adapter)
+    ]
+
+
+def _adapter_has_external_address(adapter: Adapter) -> bool:
+    """Adapter has a non-loopback and non-link-local address."""
+    return any(
+        _has_external_address(v4_config["address"]) for v4_config in adapter["ipv4"]
+    ) or any(
+        _has_external_address(v6_config["address"]) for v6_config in adapter["ipv6"]
+    )
+
+
+def _has_external_address(ip_str: str):
+    ip_addr = ip_address(ip_str)
+    return (
+        not ip_addr.is_multicast
+        and not ip_addr.is_loopback
+        and not ip_addr.is_link_local
+    )
+
+
+def load_adapters(next_hop: str | None) -> list[Adapter]:
+    """Load adapters."""
+    adapters = ifaddr.get_adapters()
+    ha_adapters: list[Adapter] = []
+    if next_hop:
+        next_hop_address = ip_address(next_hop)
+
+    for adapter in adapters:
+        ip_v4s: list[IPv4ConfiguredAddress] = []
+        ip_v6s: list[IPv6ConfiguredAddress] = []
+        default = False
+
+        for ip_config in adapter.ips:
+            if ip_config.is_IPv6:
+                try:
+                    ip_addr = ip_address(ip_config.ip[0])
+                except ValueError:
+                    continue
+                ip_v6: IPv6ConfiguredAddress = {
+                    "address": str(ip_addr),
+                    "flowinfo": ip_config.ip[1],
+                    "scope_id": ip_config.ip[2],
+                    "network_prefix": ip_config.network_prefix,
+                }
+                if next_hop and ip_addr == next_hop_address:
+                    default = True
+                ip_v6s.append(ip_v6)
+            else:
+                try:
+                    ip_addr = ip_address(ip_config.ip)
+                except ValueError:
+                    continue
+                ip_v4: IPv4ConfiguredAddress = {
+                    "address": str(ip_addr),
+                    "network_prefix": ip_config.network_prefix,
+                }
+                if next_hop and ip_addr == next_hop_address:
+                    default = True
+                ip_v4s.append(ip_v4)
+
+        ha_adapter: Adapter = {
+            "name": adapter.nice_name,
+            "enabled": False,
+            "default": default,
+            "ipv4": ip_v4s,
+            "ipv6": ip_v6s,
+        }
+        ha_adapters.append(ha_adapter)
+
+    return ha_adapters
+
+
+def _get_ip_route(dst_ip: str) -> Any:
+    """Get ip next hop."""
+    return IPRoute().route("get", dst=dst_ip)
+
+
+def _first_ip_nexthop_from_route(routes: Iterable) -> None | str:
```

Comment:
`Optional[X]` is usually translated to `X | None`.
```suggestion
def _first_ip_nexthop_from_route(routes: Iterable) -> str | None:
```